wrapper/src/main/java/software/amazon/jdbc/plugin/cache/DataRemoteCachePlugin.java

```diff
@@ -304,6 +305,10 @@ public <T, E extends Exception> T execute(
           }
```

Comment:
Maybe remove this handling of `SQLException` here because the logic is sort of redundant from your newly added logic below?